x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_high_level_stats.js

```diff
@@ -187,8 +188,75 @@ export function getHighLevelStats(server, callCluster, clusterUuids, start, end,
  * @param {String} product The product to limit too ('kibana', 'logstash', 'beats')
  * @return {Promise} Response for the instances to fetch detailed for the product.
  */
-export function fetchHighLevelStats(server, callCluster, clusterUuids, start, end, product) {
+
+
+export function getLegacyParams(server, clusterUuids, start, end, product) {
   const config = server.config();
+
+  return {
+    index: getIndexPatternForStackProduct(product),
+    size: config.get('xpack.monitoring.max_bucket_size'),
+    ignoreUnavailable: true,
+    filterPath: [
+      'hits.hits._source.cluster_uuid',
+      `hits.hits._source.${product}_stats.${product}.version`,
+      `hits.hits._source.${product}_stats.os`,
+      `hits.hits._source.${product}_stats.usage`,
+      // we don't want metadata
+      `hits.hits._source.${product}_stats.cloud.name`,
+      `hits.hits._source.${product}_stats.cloud.id`,
+      `hits.hits._source.${product}_stats.cloud.vm_type`,
+      `hits.hits._source.${product}_stats.cloud.region`,
+      `hits.hits._source.${product}_stats.cloud.zone`
+    ],
+    body: {
+      query: createQuery({
+        start,
+        end,
+        type: `${product}_stats`,
+        filters: [
+          { terms: { cluster_uuid: clusterUuids } },
+          {
+            'bool': {
```

Comment:
Filters are inherently `must` groups without the scoring and this is being embedded in a `bool / filter`, so we've already done away with the scoring bit. Logically, this query is becoming:

```
WHERE
  cluster_uuid IN clusterUuids
AND
  (
    kibana_stats.kibana.statuses.name = 'includes-usage-data'
  AND
    kibana_stats.kibana.statuses.state = 'includes-usage-data'
  )
```

We should just drop the `bool / must` here and add the two `match` queries, which likely should be `term` queries (assuming both fields are `keyword`s, which they are), directly to the filters array. Thus making it the same logical query:

```
WHERE
  cluster_uuid IN clusterUuids
AND
  kibana_stats.kibana.statuses.name = 'includes-usage-data'
AND
  kibana_stats.kibana.statuses.state = 'includes-usage-data'
```

Separately, this query seems to be treating `kibana_stats.kibana.statuses` as an associative array, which is not true at query time in ES unless you use a `nested` object and a `nested` query context (only possible with a `nested` object, which we didn't use). Ignoring that, we don't need the name -- only the value -- unless we suspect that other statuses' `state` is that same value.

Finally, these values are only going to be added to the newer documents by this PR and thus future versions where the fix exists, so this won't help with legacy documents anyway and will actually cause this query to ignore them too because they'll never have this field in them. Presumably then this was supposed to be a `must_not` rather than a `must` and my first point here should be ignored.

(This is all probably moot based on my other comment about the `range` qury`, but I wanted to make it clear)